ddtrace/tracer/tracer.go

```diff
@@ -306,16 +316,50 @@ func (t *tracer) worker(tick <-chan time.Time) {
 	}
 }
 
-func (t *tracer) pushTrace(trace []*span) {
+// traceInfo contains span list and sampling decision of the trace
+// needed to run single span sampling.
+type traceInfo struct {
```

Comment:
Generally when you find yourself using words like "data", "info", "process" or "util" in programming, I recommend questioning that choice and perhaps trying to find something better. These are words that generally apply in any situation and are not very specific. How about this?
```suggestion
type finishedTrace struct {
```
`traceInfo` sounds a bit as if it is information about a trace, whereas this is actually a finished trace with context about sampling decision.